tests/posix/common/src/nanosleep.c

```diff
@@ -120,21 +120,45 @@ ZTEST(posix_apis, test_clock_nanosleep_errors_errno)
 	zassert_equal(rem.tv_nsec, 0, "actual: %d expected: %d", rem.tv_nsec, 0);
 }
 
+/**
+ * @brief Check that a call to nanosleep has yielded executiuon for some minimum time.
+ *
+ * Check that the actual time slept is >= the total time specified by @p s (in seconds) and
+ * @p ns (in nanoseconds).
+ *
+ * @note The time specified by @p s and @p ns is assumed to be absolute (i.e. a time-point)
+ * when @p selection is set to @ref SELECT_CLOCK_NANOSLEEP. The time is assumed to be relative
+ * when @p selection is set to @ref SELECT_NANOSLEEP.
+ *
+ * @param selection Either @ref SELECT_CLOCK_NANOSLEEP or @ref SELECT_NANOSLEEP
+ * @param clock_id The clock to test (e.g. @ref CLOCK_MONOTONIC or @ref CLOCK_REALTIME)
+ * @param flags Flags to pass to @ref clock_nanosleep
+ * @param s Partial lower bound for yielded time (in seconds)
+ * @param ns Partial lower bound for yielded time (in nanoseconds)
+ */
 static void common_lower_bound_check(int selection, clockid_t clock_id, int flags, const uint32_t s,
 				     uint32_t ns)
 {
 	int r;
 	uint64_t actual_ns;
 	uint64_t exp_ns;
-	uint32_t now;
-	uint32_t then;
+	uint64_t now;
+	uint64_t then;
 	struct timespec rem = {0, 0};
 	struct timespec req = {s, ns};
 
 	errno = 0;
+#if CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER
+	then = k_cycle_get_64();
+#else
 	then = k_cycle_get_32();
+#endif
 	r = select_nanosleep(selection, clock_id, flags, &req, &rem);
+#if CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER
+	now = k_cycle_get_64();
+#else
 	now = k_cycle_get_32();
+#endif
```

Comment:
Please add 64-bit change to a separate commit.

This can go up top (let's avoid using `#ifdef` everywhere).
```cpp
static inline uint64_t cycle_get_64(void)
{
    if (IS_ENABLED(CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER) {
        return k_cycle_get_64();
    } else {
        return k_cycle_get_32();
    }
}
```

```suggestion
	then = cycle_get_64();
 	r = select_nanosleep(selection, clock_id, flags, &req, &rem);
	now = cycle_get_64();
```

Reply:
updated. Thanks for this recommendation. @cfriedt